Windows: EditorService unit tests fail — path.join() uses backslashes for mocked macOS paths

## Environment
- **OS:** Windows 11 Home (10.0.26200)
- **Node.js:** v24.14.0

## Description

9 tests in `electron/services/__tests__/EditorService.test.ts` fail on Windows. The tests mock `process.platform` as `"darwin"` but `path.join()` still uses the Windows path separator (backslash), causing every path assertion to fail.

## Steps to Reproduce

1. On Windows, run `npm run test`
2. Observe 9 failures in `EditorService.test.ts`

## Failing Tests

All 9 tests in the `EditorService.discover` suite that mock macOS paths:
- discovers JetBrains IDE via .app bundle on macOS
- discovers IntelliJ IDEA via .app bundle on macOS
- discovers JetBrains IDE in ~/Applications on macOS
- still discovers JetBrains IDE via Toolbox on macOS
- discovers VS Code via .app bundle on macOS
- discovers Sublime Text via .app bundle on macOS
- discovers new JetBrains binaries (clion, datagrip, rubymine)
- discovers editors via PATH when available
- discovers PyCharm via .app bundle on macOS

## Error

```
AssertionError: expected '\Applications\WebStorm.app\Contents\MacOS\webstorm'
              to be '/Applications/WebStorm.app/Contents/MacOS/webstorm'

AssertionError: expected '\usr\local\bin\code' to be '/usr/local/bin/code'
```

## Root Cause

`electron/services/EditorService.ts` constructs platform-specific paths using `path.join()`:

```ts
// line 144
dirs.push(path.join("/Applications", `${name}.app`, subPath));
```

On Windows, `path.join("/Applications", "WebStorm.app", ...)` returns `\Applications\WebStorm.app\...` (Windows backslash separator). The tests mock `process.platform = "darwin"` but `path.join()` is not mocked — it still uses the host OS path module, so assertions against Unix-style paths fail.

## Impact

- CI is Ubuntu-only for PRs, so this regression is invisible in automated CI
- Any Windows developer running `npm run test` locally sees 9 unexplained failures
- The actual runtime behavior on Windows is fine (these macOS code paths are guarded by `process.platform === "darwin"` checks at runtime), but test coverage is broken

## Suggested Fix

In `EditorService.ts`, use `path.posix.join()` when constructing non-Windows paths (inside `platform !== "win32"` branches), or mock `path` in the test to use `path.posix` when platform is mocked as `darwin`.

--- a/electron/ipc/handlers/terminal/__tests__/lifecycle.shellReady.test.ts
+++ b/electron/ipc/handlers/terminal/__tests__/lifecycle.shellReady.test.ts
@@ -64,12 +64,14 @@ function createEmitterPtyClient() {
 
 describe("agent command injection - shell ready detection", () => {
   const project = { id: "proj-id", name: "Project", path: process.cwd() };
+  const originalPlatform = process.platform;
 
   let ptyClient: ReturnType<typeof createEmitterPtyClient>;
   let cleanup: (() => void) | undefined;
 
   beforeEach(() => {
     vi.clearAllMocks();
+    Object.defineProperty(process, "platform", { value: "linux" });
     ptyClient = createEmitterPtyClient();
     cleanup = undefined;
     mockGetCurrentProject.mockReturnValue(project);
@@ -78,6 +80,7 @@ describe("agent command injection - shell ready detection", () => {
   });
 
   afterEach(() => {
+    Object.defineProperty(process, "platform", { value: originalPlatform });
     cleanup?.();
     ptyClient.removeAllListeners();
   });

--- a/electron/services/EditorService.ts
+++ b/electron/services/EditorService.ts
@@ -141,8 +141,8 @@ function macAppBundleDirs(apps: Array<{ name: string; subPath?: string }>): stri
   if (process.platform !== "darwin") return [];
   const dirs: string[] = [];
   for (const { name, subPath = "Contents/MacOS" } of apps) {
-    dirs.push(path.join("/Applications", `${name}.app`, subPath));
-    dirs.push(path.join(os.homedir(), "Applications", `${name}.app`, subPath));
+    dirs.push(path.posix.join("/Applications", `${name}.app`, subPath));
+    dirs.push(path.posix.join(os.homedir(), "Applications", `${name}.app`, subPath));
   }
   return dirs;
 }
@@ -151,10 +151,17 @@ function jetbrainsToolboxScriptDirs(): string[] {
   const dirs: string[] = [];
   if (process.platform === "darwin") {
     dirs.push(
-      path.join(os.homedir(), "Library", "Application Support", "JetBrains", "Toolbox", "scripts")
+      path.posix.join(
+        os.homedir(),
+        "Library",
+        "Application Support",
+        "JetBrains",
+        "Toolbox",
+        "scripts"
+      )
     );
   } else if (process.platform === "linux") {
-    dirs.push(path.join(os.homedir(), ".local", "share", "JetBrains", "Toolbox", "scripts"));
+    dirs.push(path.posix.join(os.homedir(), ".local", "share", "JetBrains", "Toolbox", "scripts"));
   } else if (process.platform === "win32") {
     const localAppData = process.env.LOCALAPPDATA || path.join(os.homedir(), "AppData", "Local");
     dirs.push(path.join(localAppData, "JetBrains", "Toolbox", "scripts"));
@@ -163,7 +170,8 @@ function jetbrainsToolboxScriptDirs(): string[] {
 }
 
 function findBinaryInPath(binary: string, extraDirs: string[] = []): string | null {
-  const pathDirs = (process.env.PATH ?? "").split(path.delimiter).filter(Boolean);
+  const p = process.platform === "win32" ? path.win32 : path.posix;
+  const pathDirs = (process.env.PATH ?? "").split(p.delimiter).filter(Boolean);
   const searchDirs = [...extraDirs, ...pathDirs];
 
   const extensions =
@@ -173,7 +181,7 @@ function findBinaryInPath(binary: string, extraDirs: string[] = []): string | nu
 
   for (const dir of searchDirs) {
     for (const ext of extensions) {
-      const fullPath = path.join(dir, binary + ext);
+      const fullPath = p.join(dir, binary + ext);
       try {
         const stat = fs.statSync(fullPath);
         if (stat.isFile()) return fullPath;

--- a/electron/services/__tests__/EditorService.test.ts
+++ b/electron/services/__tests__/EditorService.test.ts
@@ -1,4 +1,3 @@
-import path from "path";
 import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
 
 const fsMock = vi.hoisted(() => ({
@@ -36,9 +35,9 @@ describe("EditorService.discover", () => {
   });
 
   function mockExistingFiles(paths: string[]) {
-    const pathSet = new Set(paths.map((p) => path.normalize(p)));
+    const pathSet = new Set(paths);
     fsMock.statSync.mockImplementation((filePath: string) => {
-      if (pathSet.has(path.normalize(filePath))) {
+      if (pathSet.has(filePath)) {
         return { isFile: () => true };
       }
       throw new Error("ENOENT");

